### PR TITLE
feat: get-all-file-details-and-file-data

### PR DESCRIPTION
**Pull request by pallavighule**
### What ###

- Get all file details and file data

### CODE REVIEW

File: apps/api-gateway/src/issuance/issuance.service.ts [hunk 1]
```diff
@@ -162,5 +162,12 @@ export class IssuanceService extends BaseService {
         const payload = { orgId, userDetails };
         return this.sendNatsMessage(this.issuanceProxy, 'delete-issuance-records', payload);
     }
+    async getFileDetailsAndFileDataByFileId(orgId: string, fileId: string): Promise<{ response: object }> {
+        const payload = {
+          orgId,
+          fileId
+        };
+        return this.sendNats(this.issuanceProxy, 'issued-file-data-and-file-details', payload);
```
Comment (bhavanakarwade) [inline apps/api-gateway/src/issuance/issuance.service.ts:170]: Recently, we've been using the `sendNatsMessage` function. In this function, there's no need to add response as a key, you can use Promise<object> instead. Please refer to the existing functions for reference.

File: apps/api-gateway/src/issuance/issuance.controller.ts [hunk 2]
```diff
@@ -507,6 +507,39 @@ async downloadBulkIssuanceCSVTemplate(
...
+    @Param('orgId') orgId: string,
+    @Param(new ValidationPipe({ transform: true })) query: FileQuery,
+    @Res() res: Response
+  ): Promise<object> {
+    const { fileId } = query;
+    const issuedFileDetails = await this.issueCredentialService.getFileDetailsAndFileDataByFileId(orgId, fileId);
+    const finalResponse: IResponseType = {
+      statusCode: HttpStatus.OK,
+      message: ResponseMessages.issuance.success.fileDetailsAndFileData,
+      data: issuedFileDetails.response
```
Comment (bhavanakarwade) [inline apps/api-gateway/src/issuance/issuance.controller.ts:539]: Using `sendNatsMessage` which I have mentioned in service function, you can simply use data: issuedFileDetails. As part of the new API refactorization task, we no longer include the response key in the data. Please refer to other functions for reference

**Review by bhavanakarwade — CHANGES_REQUESTED** [review]

File: apps/api-gateway/src/issuance/issuance.service.ts (hunk 1 above)
Comment (pallavighule) [reply apps/api-gateway/src/issuance/issuance.service.ts]: resolved

**Review by bhavanakarwade — APPROVED** [review]
LGTM

**Pull request merged**